src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo);
+        LocalDate fromDate = LocalDate.parse(dateFrom, dateFormatter);
+        LocalDate toDate = LocalDate.parse(dateTo, dateFormatter);
+        for (String employ : names) {
+            int salary = 0;
+            for (String entry : data) {
+                String[] parts = entry.split(" ");
+                String date = parts[0];
+                String name = parts[1];
+                int hours = Integer.parseInt(parts[2]);
+                int incomePerHour = Integer.parseInt(parts[3]);
+                LocalDate entryDate = LocalDate.parse(date, dateFormatter);
+                if (!entryDate.isBefore(fromDate) && !entryDate.isAfter(toDate)) {
```

Comment:
let's move this check to the new method  compareDates() for better code readability